### PR TITLE
New summary

### PR DESCRIPTION
**Pull request by MVrachev**
I changed our summary. The messages now are making more sense.

This is fix for issues https://github.com/vmware/precaution/issues/64
 
Because there are many possible situation I wanted to make the code smaller as possible.
I hope it's readable.

I had to change besides merge_reports also the bandit and gosec report functions so they will give summary relative to the new merged summary. Now I get the amount of errors, warnings and notices from the annotations of Gosec and Bandit not from the raw summary given by Gosec and Bandit.

I added a useful function with which I count the amount of errors, warnings and notices in annotation array in annotations_levels.

Also I had to change the tests in test/merge.reports.test.js so they won't fail.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

**Comment by MVrachev** [comment]
Here are example messages:

image

when called like this:

image

**Comment by ericwb** [comment]
I don't think stating what color a warning, error, or notice is necessary.  Plus, who knows, GitHub could change the icon and color on us.  Or if someone is color blind, it also wouldn't be relevant.

File: merge_reports.js [hunk 1]
```diff
@@ -2,28 +2,42 @@
...
 const { config } = require('./config')
+/**
+ * @param {Number} errors the amount of errors found
+ * @param {Number} warnings the amount of warnings found
+ * @param {Number} notices the amount of notices found
+ */
+function getCorrectSummary (errors, warnings, notices) {
+  let summary = ''
+  let errorPrefix = errors > 1 ? 'There where ' + errors + ' errors found.' : 'There is 1 error found.'
+  let warningPrefix = warnings > 1 ? 'There where ' + warnings + ' warnings found.' : 'There is 1 warning found.'
```
Comment (ericwb) [inline merge_reports.js]:
```suggestion
  let warningPrefix = warnings > 1 ? 'There were ' + warnings + ' warnings found.' : 'There was 1 warning found.'
```

File: merge_reports.js [hunk 2]
```diff
@@ -2,28 +2,42 @@
...
+  let noticePrefix = notices > 1 ? 'There where ' + notices + ' notices found.' : 'There is 1 notices.'
```
Comment (ericwb) [inline merge_reports.js]:
```suggestion
  let noticePrefix = notices > 1 ? 'There were ' + notices + ' notices found.' : 'There was 1 notices.'
```

File: annotations_levels.js [hunk 3]
```diff
@@ -3,6 +3,23 @@
 
 const { annotationsLevels } = require('./config')
 
+function countAnnotationLevels (annotations) {
+  let errors = 0
+  let warnings = 0
+  let notices = 0
+
+  for (let i = 0; i < annotations.length; ++i) {
```
Comment (ericwb) [inline annotations_levels.js]:
```suggestion
   for (annotation in annotations) {
```

File: annotations_levels.js [hunk 4]
```diff
@@ -3,6 +3,23 @@
...
+    if (annotations[i].annotation_level === 'failure') {
```
Comment (ericwb) [inline annotations_levels.js]:
```suggestion
     if (annotation.annotation_level === 'failure') {
```

File: annotations_levels.js [hunk 5]
```diff
@@ -3,6 +3,23 @@
...
+      errors += 1
+    } else if (annotations[i].annotation_level === 'warning') {
```
Comment (ericwb) [inline annotations_levels.js]:
```suggestion
     } else if (annotation.annotation_level === 'warning') {
```

**Comment by evqna** [comment]
This still feels too wordy to me, I would rather simply have something such as: "Results: *n* Errors, *m* Warnings, *p* Notices".

**Comment by MVrachev** [comment]
But that way there is clear separation from erros, notices and warnings.
Also that way we can put emojis.

**Comment by evqna** [comment]
Well there's no need to have it all in one line, you can have a list like:
- ❌ 2 errors
- ⚠️ 5 warnings
- ℹ️ 3 notices
or whatever symbols you want to use
